cmd/telliot/cmds.go

```diff
@@ -0,0 +1,316 @@
+package main
+
+import (
+	"context"
+	"math/big"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/alecthomas/kong"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	tellorCommon "github.com/tellor-io/telliot/pkg/common"
+	"github.com/tellor-io/telliot/pkg/config"
+	"github.com/tellor-io/telliot/pkg/contracts"
+	"github.com/tellor-io/telliot/pkg/db"
+	"github.com/tellor-io/telliot/pkg/ops"
+	"github.com/tellor-io/telliot/pkg/rpc"
+	"github.com/tellor-io/telliot/pkg/util"
+)
+
+type configPath string
+
+func (c configPath) AfterApply(kong *kong.Context) error {
+	err := config.ParseConfig(string(c))
+	if err != nil {
+		return errors.Wrapf(err, "parsing config")
+	}
+
+	cfg := config.GetConfig()
+	ctx := context.Background()
+	err = util.SetupLoggingConfig(cfg.Logger)
+	if err != nil {
+		return errors.Wrapf(err, "parsing log config")
+	}
+
+	logger := util.SetupLogger(cfg.LogLevel)
+
+	client, contract, account, err := setup(ctx, cfg)
+	if err != nil {
+		return errors.Wrapf(err, "setting up variables")
+	}
+
+	kong.BindTo(logger, (*log.Logger)(nil))
+	kong.BindTo(client, (*rpc.ETHClient)(nil))
+	kong.BindTo(ctx, (*context.Context)(nil))
+	kong.Bind(contract)
+	kong.Bind(account)
+	kong.Bind(cfg)
```

Comment:
This looks very suspiciously to a global var approach. Unless there is blocker I think we should just create these within the `Run(` function instead of this interesting `Bind` magic.

Reply:
Hmm I agree it looks suspicious. I think there's two negatives of creating this in run functions:
1. It'll lead to a lot of repetition, every Run func will need to have this logic for parsing and executing the config file.
2. I think it'll be more verbose to write the logic to actually finding the config path, as I'm not sure if flags are easily accessible within run commands.

On the other hand, this global context is only used to pass arguments from setup to designated run functions, from there everything is explicitly passed through arguments, which is why I didn't bother too much.

But let me know what you think!

Reply:
> It'll lead to a lot of repetition, every Run func will need to have this logic for parsing and executing the config file.

I think this is fine. We can extract the repeatable part in functions to reduce the repetition. Most golang devs(myself included) prefer to no magic whatsoever. I am not sure why the cli dev decided to use this global variables pattern.

> I think it'll be more verbose to write the logic to actually finding the config path, as I'm not sure if flags are easily accessible within run commands.

I don't understand the problem here. can you ping me and can discuss. 
This is what I imagine when you have a command that needs a config `mine --config-file=...` Inside run it should have access to the `config-file` variable right? This way it will just parse the config and build all the other required structs components etc.